Assets/Scripts/Lazo.cs

```diff
@@ -0,0 +1,136 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Laz
+{
+    public class Lazo
+    {
+        private List<LazoPosition> _listOfPositions = new List<LazoPosition>();
+        private ILazoProperties _lazoProperties = null;
+
+        private bool _isLazoing = false;
+        private float _rateOfRecordingTimerElapsed = 0;
+
+        public bool IsLazoing
+        {
+            get => _isLazoing;
+            set
+            {
+                _isLazoing = value;
+                if (!_isLazoing)
+                {
+                    Clear();
+                    // Debug
+                    DebugClearListOfCubes();
+                }
+            }
+        }
+
+        public Lazo(ILazoProperties properties)
+        {
+            _lazoProperties = properties;
+        }
+
+        /// <summary>
+        /// Storing new positions that player moved to if able to depending on timer
+        /// </summary>
+        /// <param name="position">position</param>
+        public void RunLazoIfAble(Vector3 position, float deltaTime)
+        {
+            _rateOfRecordingTimerElapsed -= deltaTime;
+            if (_rateOfRecordingTimerElapsed < 0)
+            {
+                RunLazo(position);
+                ResetRateOfRecordingTimeElapsed();
+            }
+
+            RemoveOldestPointIfNeeded(deltaTime);
+        }
+
+        private void RemoveOldestPointIfNeeded(float deltaTime)
+        {
+            _listOfPositions = _listOfPositions.Select(lazoPosition =>
```

Comment:
You can imagine this section as a mapping function
- meaning - for each LazoPosition...do all this underneath.